src/components/Csam/CsamDetailsForm.tsx

```diff
@@ -0,0 +1,198 @@
+import {
+  Checkbox,
+  Radio,
+  Stack,
+  Center,
+  Loader,
+  Alert,
+  Card,
+  Input,
+  Group,
+  Text,
+  Badge,
+  Button,
+  Container,
+} from '@mantine/core';
+import { IconExternalLink, IconPhoto } from '@tabler/icons-react';
+import { uniqBy } from 'lodash';
+
+import { useEffect, useMemo } from 'react';
+import { z } from 'zod';
+import { useCsamContext } from '~/components/Csam/CsamProvider';
+import {
+  useCsamImageSelectStore,
+  useCsamModelVersionSelectStore,
+} from '~/components/Csam/useCsamImageSelect.store';
+import { PopConfirm } from '~/components/PopConfirm/PopConfirm';
+import { ScrollArea } from '~/components/ScrollArea/ScrollArea';
+import { Stepper, useStepperContext } from '~/components/Stepper/Stepper';
+import { Form, InputCheckboxGroup, InputRadioGroup, useForm } from '~/libs/form';
+import {
+  csamCapabilitiesDictionary,
+  csamContentsDictionary,
+  csamReportFormSchema,
+} from '~/server/schema/csam.schema';
+import { trpc } from '~/utils/trpc';
+import { isDefined } from '~/utils/type-guards';
+
+export function CsamDetailsForm() {
+  const { userId, isInternal } = useCsamContext();
+  const { next } = useStepperContext();
+  const form = useForm({
+    schema: csamReportFormSchema,
+    defaultValues: { origin: isInternal ? 'testing' : 'user' },
+    shouldUnregister: false,
+  });
+
+  const { mutate: createReport, isLoading } = trpc.csam.createReport.useMutation();
+
+  const handleSubmit = (data: z.infer<typeof csamReportFormSchema>) => {
+    createReport({
+      ...data,
+      userId,
+      modelVersionIds: useCsamModelVersionSelectStore.getState().getSelected(userId),
+      images: useCsamImageSelectStore
+        .getState()
+        .getSelected(userId)
+        .map((id) => ({ id })),
+    });
+    next();
```

Comment:
Should we wait for the request to be successful before moving to the next step? Also, we should show some notification in case the request fails too